tilecloud_chain/timedtilestore.py

```diff
@@ -0,0 +1,71 @@
+from c2cwsgiutils import stats
+from tilecloud import TileStore
+
+
+class TimedTileStoreWrapper(TileStore):
+    """
+    A wrapper around a TileStore that adds timer metrics.
+    """
+    def __init__(self, tile_store, stats_name):
+        super().__init__()
+        self._tile_store = tile_store
+        self._stats_name = stats_name
+
+    def _get_stats_name(self, func_name, tile=None):
+        if tile and 'layer' in tile.metadata:
+            return [self._stats_name, tile.metadata['layer'], func_name]
+        else:
+            return [self._stats_name, func_name]
+
+    def _time_iteration(self, generator, func_name):
+        while True:
+            try:
+                timer = stats.timer()
```

Comment:
Why not using the `_time_iteration...`?

Reply:
In `_time_iteration` itself?

Reply:
Because I don't know the name of the metric in advance.